tests/integrated/test_suite

```diff
@@ -58,6 +64,113 @@ try:
+# certain amount of threads that we are told to use, we try to run
+# so many tests in parallel, that the total core count is not higher
+# than this number. If this isn't possible because some jobs require
+# more cores than we have threads, we run the remaining tests in
+# serial. The number of parallel threads is called cost of a test.
+#
+# The beginning of MAKEFLAGS has some flags that tell us what we are
+# supposed to be doing. Most of them we can ignore.
+# Check what we are supposed to be doing:
+if 'MAKEFLAGS' in os.environ:
+    makeflags = os.environ['MAKEFLAGS']
+    for token in makeflags:
+        if token == "n":
+            # Print recipies:
+            # We ignore this for now ...
+            sys.exit(0)
+        elif token == "q":
+            # Question mode:
+            # We are never up-to-date
+            sys.exit(1)
+        elif token in "ksBd":
+            # k - keep going
+            # s - silent
+            # B - always build (we do)
+            # d - debug
+            pass
+        elif token == " ":
+            break
+        elif token == "-":
+            # This is not from make
+            break
+        else:
+            # Not implemented
+            print("mode '%s' not implemented"%token)
+            #sys.exit(42)
+else:
+    makeflags = ""
+num_threads=1
+if args.jobs:
+    num_threads = args.jobs[0]
+js_read=None
+# if we are running under make, we only have one thread (called job
+# in make terms) for sure, but might be able to run more threads
+# (depending what other jobs are currently running)
+# Additional jobs come in form of `tokens` that we need to give
+# back, if we are finished.
+# https://www.gnu.org/software/make/manual/html_node/POSIX-Jobserver.html
+if "--jobserver-auth=" in makeflags or "--jobserver-fds=" in makeflags:
+    try:
+        offset=makeflags.index("--jobserver-auth=")
+    except ValueError:
+        offset=makeflags.index("--jobserver-fds=")
+    file_desc=[int(i) for i in re.findall("\d+",makeflags[offset:])[0:2]]
+    js_read = file_desc[0]
+    js_write = file_desc[1]
+    js_tokens = b""
+
+class Test(threading.Thread):
+    def __init__(self, name, cmd):
+        threading.Thread.__init__(self)
+        self.name = name
+        self.cmd = cmd
+        self.local = threading.local()
+        self.cost = self._cost()
+    def run(self):
+        # Check requirements
+        self.local.req_met, self.local.req_expr = requirements.check(self.name+"/runtest")
+        self.local.start_time = time.time()
+        if not self.local.req_met:
+            print(output % self.name +
+                  "S - {0} => False".format(self.local.req_expr))
+            sys.stdout.flush()
+            self.status = 0
+        else:
+            # Run test, piping stdout so it is not sent to console
+            self.status, self.output = shell("cd "+self.name+";"+self.cmd, pipe=True)
+            print(output % self.name, end='')
```

Comment:
Should this be `self.output`? It seems to work, but it's very much not clear where it's coming from

Reply:
It is defined in line 250:
https://github.com/boutproject/BOUT-dev/pull/2000/files?file-filters%5B%5D=.py&file-filters%5B%5D=No+extension#diff-fb8b5c428df66ca9d6c9af2419516cceL104

I am happy to change it to avoid confusion between `output` and `self.output`.
The first is the formatting string with the length, the second is the output from the run ...

Reply:
Ah I see! Maybe instead pass in `longest` as an argument to `__init__` with a name like `width`? Then something like `"{:{width}}".format(self.name, width=self.width)`, and similar for L141. Currently it's a bit too "magic" for me